From 86686ccbf43c5d9e8c8dc97c66ba09e522050e5e Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Thu, 27 Jul 2017 10:06:24 +0800
Subject: [PATCH 1/3] fix configure and compile failures

1. Fix do_configure failure
---------------------------
|Checking header volume_key/libvolume_key.h existence and usability.
../tmp/6tvtK.c:1:38: fatal error: volume_key/libvolume_key.h:
No such file or directory
| #include <volume_key/libvolume_key.h>
|Checking header dmraid/dmraid.h existence and usability.../tmp/
ktVJ6.c:1:27: fatal error: dmraid/dmraid.h: No such file or directory
| #include <dmraid/dmraid.h>
---------------------------
We explictly add volume_key and dmraid to DEPENDS, do not need
configure to test.

2. Fix config.h not found
Add it to configure.ac

3. Correct AC_DEFINE
...
autoheader: warning: missing template: LIBMOUNT_NEW_ERR_API
autoheader: Use AC_DEFINE([LIBMOUNT_NEW_ERR_API], [], [Description])
...

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 configure.ac | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/configure.ac b/configure.ac
index a37f337..e9a4964 100644
--- a/configure.ac
+++ b/configure.ac
@@ -9,6 +9,8 @@ AC_DISABLE_STATIC
 AM_INIT_AUTOMAKE([foreign -Wall -Werror -Wno-syntax -Wno-portability])
 AC_CONFIG_MACRO_DIR([m4])
 
+AC_CONFIG_HEADERS([config.h])
+
 AM_PATH_PYTHON
 
 AM_PROG_AR
@@ -194,7 +196,8 @@ LIBBLOCKDEV_PKG_CHECK_MODULES([KMOD], [libkmod >= 19])
 AS_IF([test "x$with_crypto" != "xno"],
       [LIBBLOCKDEV_PKG_CHECK_MODULES([CRYPTSETUP], [libcryptsetup >= 1.6.7])
       AS_IF([$PKG_CONFIG --atleast-version=2.0 libcryptsetup],
-            [AC_DEFINE([LIBCRYPTSETUP_2])], [])
+            [AC_DEFINE([LIBCRYPTSETUP_2])], [1], [Version libcryptsetup >= 2],
+            [])
       AS_IF([test "x$with_escrow" != "xno"],
             [LIBBLOCKDEV_PKG_CHECK_MODULES([NSS], [nss >= 3.18.0])
              LIBBLOCKDEV_CHECK_HEADER([volume_key/libvolume_key.h], [$GLIB_CFLAGS $NSS_CFLAGS], [libvolume_key.h not available])],
@@ -206,10 +209,6 @@ AS_IF([test "x$with_dm" != "xno" -o "x$with_lvm" != "xno" -o "x$with_lvm_dbus" !
       [LIBBLOCKDEV_PKG_CHECK_MODULES([DEVMAPPER], [devmapper >= 1.02.93])],
       [])
 
-AS_IF([test "x$with_dm" != "xno" -a "x$with_dmraid" != "xno"],
-      [LIBBLOCKDEV_CHECK_HEADER([dmraid/dmraid.h], [], [dmraid.h not available])],
-      [])
-
 AS_IF([test "x$with_part" != "xno" -o "x$with_fs" != "xno"],
       [LIBBLOCKDEV_PKG_CHECK_MODULES([PARTED], [libparted >= 3.1])]
       [])
@@ -218,7 +217,8 @@ AS_IF([test "x$with_fs" != "xno"],
       [LIBBLOCKDEV_PKG_CHECK_MODULES([MOUNT], [mount >= 2.23.0])
        # new versions of libmount has some new functions we can use
        AS_IF([$PKG_CONFIG --atleast-version=2.30.0 mount],
-             [AC_DEFINE([LIBMOUNT_NEW_ERR_API])], [])
+             [AC_DEFINE([LIBMOUNT_NEW_ERR_API], [1], [new versions of libmount has some new functions we can use])],
+             [])
 
        LIBBLOCKDEV_PKG_CHECK_MODULES([BLKID], [blkid >= 2.23.0])
        # older versions of libblkid don't support BLKID_SUBLKS_BADCSUM so let's just
@@ -243,7 +243,7 @@ AS_IF([test "x$with_nvdimm" != "xno"],
        LIBBLOCKDEV_PKG_CHECK_MODULES([NDCTL], [libndctl >= 58.4])
        # new versions of libndctl new modes
        AS_IF([$PKG_CONFIG --atleast-version=60 libndctl],
-             [AC_DEFINE([LIBNDCTL_NEW_MODES])], [])]
+             [AC_DEFINE([LIBNDCTL_NEW_MODES])], [libndctl >= version 60])]
       [])
 
 AS_IF([test "x$with_vdo" != "xno"],
-- 
2.14.4

